From 0474535acfde6adec1628cf02c67a296f4f38c5e Mon Sep 17 00:00:00 2001
From: Alexey Dobriyan <adobriyan@openvz.org>
Date: Mon, 24 Mar 2008 14:11:16 +0300
Subject: [PATCH 45/48] VE: fix uevent generation in VE

There is only one uevent netlink socket with ->net being init_net.
There are multiple listeners (->mc_list) both from VE0 and VE.

So, when, e. g., init scripts inside VE do echo 1 >/sys/.../uevent
uevent is discarded due to "->net doesn't match" check!

http://bugzilla.openvz.org/show_bug.cgi?id=840
---
 net/netlink/af_netlink.c |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/net/netlink/af_netlink.c b/net/netlink/af_netlink.c
index 2bd6026..ad02fab 100644
--- a/net/netlink/af_netlink.c
+++ b/net/netlink/af_netlink.c
@@ -956,9 +956,6 @@ static inline int do_one_broadcast(struct sock *sk,
 	    !test_bit(p->group - 1, nlk->groups))
 		goto out;
 
-	if ((sk->sk_net != p->net))
-		goto out;
-
 	if (!ve_accessible_strict(get_exec_env(), sk->owner_env))
 		goto out;
 
-- 
1.5.4.3

